packages/react-aria-components/src/Collection.tsx

```diff
@@ -927,10 +927,22 @@ export function Collection<T extends object>(props: CollectionProps<T>): JSX.Ele
 
   let doc = useContext(CollectionDocumentContext);
   if (doc) {
-    return <CollectionRoot>{children}</CollectionRoot>;
+    // TODO: operates off the assumption that dependencies set on a higher level Collection should propagate down
+    // Makes it simpler for someone to setup nested Tree loaders (only need to set dependency at top level Collection instead of remembering to propagate it to every instance of Collection)
+    // TODO: With the refactor to Collections, I can either apply the CollectionContext.Provider here or in Tree
+    // itself in the content passed to CollectionBuilder. Feels like here is more appropriate so we don't need to remember to do this for each component.
+    return (
```

Comment:
What does this mean for a menu inside a tree?

Reply:
I think I also added this in the move to the new package in #6640 fyi

Reply:
> What does this mean for a menu inside a tree?

It might still need to invalidate. For example, say you have something like this:

```tsx
<Tree dependencies={[something]}>
  {item => (
    <TreeItem>
      <TreeItemContent>
        {item.name}
        <MenuTrigger>
            <Button>...</Button>
            <Menu items={menuItems}>
              {menuItem => <MenuItem>{menuItem.name} • {something}</MenuItem>}
            </Menu>
        </MenuTrigger>
      </TreeItemContent>
    </TreeItem>
  )}
</Tree>
```

In this case, when `something` changes in the outer scope, we need to invalidate everything that accesses it. That means we need to re-render the Menu item. We could put `dependencies` on the `Menu`, but that wouldn't work because the `TreeItem` containing it still won't re-render. So we need to move it up to the `Tree`. Currently you'd have to put it on both Tree and Menu. I think Daniel was trying to make it a bit easier so you'd only need to set dependencies at the top-level.